# Поиск признаков успешной игры

Учебный проект по специальности "аналитик данных", 10-я неделя обучения.

Тема: итоговый проект по базовым навыкам анализа данных.

Автор: Александр Лагутин.

Начат: 6 июня 2022 г в 07:20.
Завершён: 1 июня 2022 г в 10:45.

Время подготовки: 1 рабочих часов (1 рабочий день).

<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Описание-проекта" data-toc-modified-id="Описание-проекта-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Описание проекта</a></span><ul class="toc-item"><li><span><a href="#Заказчик" data-toc-modified-id="Заказчик-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Заказчик</a></span></li><li><span><a href="#Задача" data-toc-modified-id="Задача-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Задача</a></span></li><li><span><a href="#Входные-данные" data-toc-modified-id="Входные-данные-1.3"><span class="toc-item-num">1.3&nbsp;&nbsp;</span>Входные данные</a></span><ul class="toc-item"><li><span><a href="#Термины" data-toc-modified-id="Термины-1.3.1"><span class="toc-item-num">1.3.1&nbsp;&nbsp;</span>Термины</a></span></li><li><span><a href="#Описание-данных" data-toc-modified-id="Описание-данных-1.3.2"><span class="toc-item-num">1.3.2&nbsp;&nbsp;</span>Описание данных</a></span></li></ul></li><li><span><a href="#План-работы" data-toc-modified-id="План-работы-1.4"><span class="toc-item-num">1.4&nbsp;&nbsp;</span>План работы</a></span></li><li><span><a href="#Инструменты" data-toc-modified-id="Инструменты-1.5"><span class="toc-item-num">1.5&nbsp;&nbsp;</span>Инструменты</a></span><ul class="toc-item"><li><span><a href="#Импорт-библиотек" data-toc-modified-id="Импорт-библиотек-1.5.1"><span class="toc-item-num">1.5.1&nbsp;&nbsp;</span>Импорт библиотек</a></span></li><li><span><a href="#Определения-констант" data-toc-modified-id="Определения-констант-1.5.2"><span class="toc-item-num">1.5.2&nbsp;&nbsp;</span>Определения констант</a></span></li><li><span><a href="#Определения-команд" data-toc-modified-id="Определения-команд-1.5.3"><span class="toc-item-num">1.5.3&nbsp;&nbsp;</span>Определения команд</a></span></li><li><span><a href="#Определения-чистых-функций" data-toc-modified-id="Определения-чистых-функций-1.5.4"><span class="toc-item-num">1.5.4&nbsp;&nbsp;</span>Определения чистых функций</a></span></li></ul></li></ul></li><li><span><a href="#Обзор-и-исправление-данных" data-toc-modified-id="Обзор-и-исправление-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Обзор и исправление данных</a></span><ul class="toc-item"><li><span><a href="#Итоги-обзора-и-исправления-данных" data-toc-modified-id="Итоги-обзора-и-исправления-данных-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Итоги обзора и исправления данных</a></span></li></ul></li><li><span><a href="#Расходы-каждого-пользователя-в-месяц" data-toc-modified-id="Расходы-каждого-пользователя-в-месяц-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Расходы каждого пользователя в месяц</a></span><ul class="toc-item"><li><span><a href="#Итоги-подготовки-информации-о-расходах:" data-toc-modified-id="Итоги-подготовки-информации-о-расходах:-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Итоги подготовки информации о расходах:</a></span></li></ul></li><li><span><a href="#Анализ-расходов" data-toc-modified-id="Анализ-расходов-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Анализ расходов</a></span><ul class="toc-item"><li><span><a href="#Итоги-анализа-расходов" data-toc-modified-id="Итоги-анализа-расходов-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Итоги анализа расходов</a></span></li></ul></li><li><span><a href="#Гипотезы-и-проверки" data-toc-modified-id="Гипотезы-и-проверки-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Гипотезы и проверки</a></span><ul class="toc-item"><li><span><a href="#Итоги-проверки-гипотез" data-toc-modified-id="Итоги-проверки-гипотез-5.1"><span class="toc-item-num">5.1&nbsp;&nbsp;</span>Итоги проверки гипотез</a></span></li></ul></li><li><span><a href="#Общие-итоги" data-toc-modified-id="Общие-итоги-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Общие итоги</a></span><ul class="toc-item"><li><span><a href="#Рекомендации" data-toc-modified-id="Рекомендации-6.1"><span class="toc-item-num">6.1&nbsp;&nbsp;</span>Рекомендации</a></span></li></ul></li></ul></div>

## Описание проекта

### Заказчик

Интернет-магазин "Стримчик".

Основная деятельность: продажа компьютерных игр по всему миру.

### Задача

Выявить закономерности, которые определяют успешность игры.

Для чего: чтобы магазин выбрал потенциально популярный продукт, закупил и рекламировал именно его.

### Входные данные

Сводные данные об играх. Взяты из открытых источников. Данные за последний год могут быть неполными.

#### Термины

**ESRB** (Entertainment Software Rating Board) — ассоциация, определяющая возрастной рейтинг компьютерных игр. ESRB оценивает игровой контент и присваивает ему подходящую возрастную категорию, например, «Для взрослых», «Для детей младшего возраста» или «Для подростков».

**TBD** - "to be determined". Эта аббревиатура стоит вместо величины рейтинга или оценки, когда данных недостаточно для подсчёта величины по правилам.

#### Описание данных

- Name — название игры
- Platform — платформа
- Year_of_Release — год выпуска
- Genre — жанр игры
- NA_sales — продажи в Северной Америке (миллионы проданных копий)
- EU_sales — продажи в Европе (миллионы проданных копий)
- JP_sales — продажи в Японии (миллионы проданных копий)
- Other_sales — продажи в других странах (миллионы проданных копий)
- Critic_Score — оценка критиков (максимум 100)
- User_Score — оценка пользователей (максимум 10)
- Rating — категория от ESRB

### План работы

- [ ] Загрузить и изучить общую информацию.
- [ ] Подготовьте данные:
  - [ ] Замените названия столбцов (приведите к нижнему регистру);
  - [ ] Преобразуйте данные в нужные типы. Опишите, в каких столбцах заменили тип данных и почему;
  - [ ] Обработайте пропуски при необходимости:
    - [ ] Объясните, почему заполнили пропуски определённым образом или почему не стали это делать;
    - [ ] Опишите причины, которые могли привести к пропускам;
    - [ ] Обратите внимание на аббревиатуру 'tbd' в столбце с оценкой пользователей. Отдельно разберите это значение и опишите, как его обработать;
    - [ ] Посчитайте суммарные продажи во всех регионах и запишите их в отдельный столбец.
- [ ] Шаг 3. Проведите исследовательский анализ данных
  - [ ] Посмотрите, сколько игр выпускалось в разные годы. Важны ли данные за все периоды?
  - [ ] Посмотрите, как менялись продажи по платформам. Выберите платформы с наибольшими суммарными продажами и постройте распределение по годам. За какой характерный срок появляются новые и исчезают старые платформы?
  - [ ] Возьмите данные за соответствующий актуальный период. Актуальный период определите самостоятельно в результате исследования предыдущих вопросов. Основной фактор — эти данные помогут построить прогноз на 2017 год.
  - [ ] Не учитывайте в работе данные за предыдущие годы.
  - [ ] Какие платформы лидируют по продажам, растут или падают? Выберите несколько потенциально прибыльных платформ.
  - [ ] Постройте график «ящик с усами» по глобальным продажам игр в разбивке по платформам. Опишите результат.
  - [ ] Посмотрите, как влияют на продажи внутри одной популярной платформы отзывы пользователей и критиков. Постройте диаграмму рассеяния и посчитайте корреляцию между отзывами и продажами. Сформулируйте выводы.
  - [ ] Соотнесите выводы с продажами игр на других платформах.
  - [ ] Посмотрите на общее распределение игр по жанрам. Что можно сказать о самых прибыльных жанрах? Выделяются ли жанры с высокими и низкими продажами?
- [ ] Шаг 4. Составьте портрет пользователя каждого региона
  - [ ] Определите для пользователя каждого региона (NA, EU, JP):
    - [ ] Самые популярные платформы (топ-5). Опишите различия в долях продаж.
    - [ ] Самые популярные жанры (топ-5). Поясните разницу.
    - [ ] Влияет ли рейтинг ESRB на продажи в отдельном регионе?
- [ ] Шаг 5. Проверьте гипотезы
  - [ ] Средние пользовательские рейтинги платформ Xbox One и PC одинаковые;
  - [ ] Средние пользовательские рейтинги жанров Action (англ. «действие», экшен-игры) и Sports (англ. «спортивные соревнования») разные.
  - [ ] Поясните:
    - [ ] Как вы сформулировали нулевую и альтернативную гипотезы;
    - [ ] Какой критерий применили для проверки гипотез и почему.
- [ ] Общий вывод

### Инструменты

Используем [Anaconda](https://www.anaconda.com/), окружение `ds_da_practicum_env.yml` "для Mac OS" (от 2022-03-23).

#### Импорт библиотек

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as st
import matplotlib.pyplot as plt

#### Определения констант

In [2]:
FIGSIZE = (12, 6)

#### Определения чистых функций

In [3]:
def headline(line, width=50):
    """Format plain text headline
    """
    return f'\n{line.upper():=^{width}}'

In [4]:
def count_nans(data):
    """Count NaN values in all columns
    """
    nans = []
    for column in data.columns:
        missing = data[column].isna().sum()
        part = missing / len(data)
        uniques = data[column].sort_values().unique()
        nans.append([column, part, len(uniques)])

    return pd.DataFrame(
        data=nans, columns=['column', 'na_part', 'unique_count']
    ).sort_values(['na_part', 'unique_count'], ascending=False).set_index('column')

#### Определения команд

In [5]:
def load(tablename, ext='csv'):
    """Load a table from file
    """
    filename = tablename + '.' + ext
    try:
        return pd.read_csv(filename)
    except FileNotFoundError:
        return pd.read_csv('https://code.s3.yandex.net/datasets/' + filename)

In [6]:
def survey(data):
    """Display more dataset info
    """
    display(data)
    print(f'Полностью совпадающих строк: {data.duplicated().sum()}\n')
    data.info()

In [7]:
def print_filled_rows_share(df):
    share = 1 - df.isna().any(axis=1).sum() / len(df)
    print(f'Заполненных строк без пропусков: {share:.02%}')

In [8]:
# можно было бы рисовать `bar` в случаях, когда вариантов мало.
# однако по неясной причине `bar` не рисуется, и не вызывает ошибки,
# а вызывает лишь зацикливание при построении первого же графика

def survey_columns(df):
    for column in nans_before.index:
        print(headline(column))

        uniques = realty[column].sort_values().unique()
        if len(uniques) < 20:
            print(uniques)
        print(f'Data type: {str(realty[column].dtype):>10}')
        print(f'Unique values: {len(uniques):6}')

        missing = realty[column].isna().sum()
        if missing:
            print(f'Not avaliable: {nans_before.loc[column, "na_part"]:.02%}')
        else:
            print('All values is avaliable.')

        try:
            realty.plot(
                kind='hist',
                y=column,
               bins=50,
            )
            plt.show()
        except Exception as e:
            print(str(e).capitalize())

In [9]:
def fillnaby(data, column, groupby, func='median'):
    data[column] = data[column].fillna(
        data.groupby(groupby)[column].transform(func))

In [10]:
# формирует и выводит график, подстроенный под наши нужды
def histf(data, column, show=True, bins=100, qtl=1, start=None,
          stop=None, figsize=FIGSIZE, **kwargs):
    fixed = {
        'y': column,
        'kind': 'hist',
        'bins': min(bins, len(data[column].unique())),
        'figsize': figsize,
        'title': column,
        'legend': False,
        'ec': 'black',
    }
    if 'range' not in kwargs.keys():
        if not start:
            start = data[column].min()
        if not stop:
            stop = data[column].quantile(qtl)
        kwargs['range'] = (start, stop)
    data.plot(**fixed, **kwargs)
    if show:
        plt.show()

def histf2(data, column1, column2, **kwargs):
    histf(column1, data=data, show=False, **kwargs)
    histf(column2, data=data, show=False, **kwargs)
    plt.show()

# два графика: все случаи, и те, что менее указанного квантиля
def dhistf(column, data, qtl=0.975, figsize=FIGSIZE, **kwargs):
    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=figsize)
    fig.suptitle('This is a somewhat long figure title', fontsize=16)
    
    ax1.set_title('full_data')
    ax2.set_title(f'{qtl:.2%} quantile')
    histf(column, data=data, ax=ax1, subplots=True, show=False, **kwargs)
    histf(column, data=data, ax=ax2, subplots=True, show=False, qtl=qtl, **kwargs)
    plt.show()
# Развитая форма этого чудовища должна принимать список квантилей,
# и строить столько графиков, сколько квантилей указано,
# и качестве центра диапазона выбирать медиану.
# Пора перейти к аналитике.

In [11]:
def typical(data, column, ratio=1.5):
    q1 = data[column].quantile(0.25)
    q3 = data[column].quantile(0.75)
    iqr = q3 - q1
    whiskers = iqr * ratio
    low = q1 - whiskers
    high = q3 + whiskers
    
    print(f'{column.title()}: min = {data[column].min()}, max = {data[column].max()}, normal range = [{low:.2f} ... {high:.2f}]')
    return data.query(f'@low <= {column} <= @high')

In [12]:
quantitative_properties = [
    'last_price_mln', 'total_area', 'living_area_share', 'kitchen_area_share',
    'ceiling_height', 'city_centers_nearest', 'airports_nearest',
]

categorical_properties = [
    'rooms', 'balcony', 'floor_category',
    'is_apartment', 'is_studio', 'has_open_plan',
    'parks_around3000', 'ponds_around3000',
    'weekday', 'month', 'year',
]

def batch_corr(data, column='last_price_mln'):
    print(headline('Коэффициенты линейной корреляции'))
    display(
            data
            .reindex(columns=(quantitative_properties + categorical_properties))
            .corr()
            [column].sort_values(ascending=False)
    )

def batch_scatter(data, column='last_price_mln'):
    for prop in quantitative_properties:
        if prop == column:
            continue
        data.plot(
            kind='scatter', alpha=0.2, figsize=FIGSIZE,
            y=column,
            x=prop,
        )
    plt.show()

def batch_box(data, column='last_price_mln', by_props=categorical_properties):
    for prop in by_props:
        data.boxplot(
            column=column,
            by=prop,
            figsize=FIGSIZE,
        )
    plt.show()

## Обзор и подготовка данных

### Загрузка данных

In [13]:
games = load('games')
survey(games)

Name Platform  Year_of_Release         Genre  \
0                         Wii Sports      Wii           2006.0        Sports   
1                  Super Mario Bros.      NES           1985.0      Platform   
2                     Mario Kart Wii      Wii           2008.0        Racing   
3                  Wii Sports Resort      Wii           2009.0        Sports   
4           Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
...                              ...      ...              ...           ...   
16710  Samurai Warriors: Sanada Maru      PS3           2016.0        Action   
16711               LMA Manager 2007     X360           2006.0        Sports   
16712        Haitaka no Psychedelica      PSV           2016.0     Adventure   
16713               Spirits & Spells      GBA           2003.0      Platform   
16714            Winning Post 8 2016      PSV           2016.0    Simulation   

       NA_sales  EU_sales  JP_sales  Other_sales  Critic_Score User_Score  \
0         41.36     28.96      3.77         8.45          76.0          8   
1         29.08      3.58      6.81         0.77           NaN        NaN   
2         15.68     12.76      3.79         3.29          82.0        8.3   
3         15.61     10.93      3.28         2.95          80.0          8   
4         11.27      8.89     10.22         1.00           NaN        NaN   
...         ...       ...       ...          ...           ...        ...   
16710      0.00      0.00      0.01         0.00           NaN        NaN   
16711      0.00      0.01      0.00         0.00           NaN        NaN   
16712      0.00      0.00      0.01         0.00           NaN        NaN   
16713      0.01      0.00      0.00         0.00           NaN        NaN   
16714      0.00      0.00      0.01         0.00           NaN        NaN   

      Rating  
0          E  
1        NaN  
2          E  
3          E  
4        NaN  
...      ...  
16710    NaN  
16711    NaN  
16712    NaN  
16713    NaN  
16714    NaN  

[16715 rows x 11 columns]

Полностью совпадающих строк: 0

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16715 entries, 0 to 16714
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16713 non-null  object 
 1   Platform         16715 non-null  object 
 2   Year_of_Release  16446 non-null  float64
 3   Genre            16713 non-null  object 
 4   NA_sales         16715 non-null  float64
 5   EU_sales         16715 non-null  float64
 6   JP_sales         16715 non-null  float64
 7   Other_sales      16715 non-null  float64
 8   Critic_Score     8137 non-null   float64
 9   User_Score       10014 non-null  object 
 10  Rating           9949 non-null   object 
dtypes: float64(6), object(5)
memory usage: 1.4+ MB


Видим, что потребуется:
- заполнить пропуски,
- переименовать колонки под lower_snake_case,
- установить для года выхода и, возможно, оценки критиков тип данных `int`,
- а для оценки пользователей - `float`.

### Замена имён колонок

In [14]:
games.columns

Index(['Name', 'Platform', 'Year_of_Release', 'Genre', 'NA_sales', 'EU_sales',
       'JP_sales', 'Other_sales', 'Critic_Score', 'User_Score', 'Rating'],
      dtype='object')

In [15]:
games.columns = [column.lower() for column in games.columns]
display(games.columns)

Index(['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales',
       'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating'],
      dtype='object')

### Преобразования типов

In [17]:
games['year_of_release'].unique()

array([2006., 1985., 2008., 2009., 1996., 1989., 1984., 2005., 1999.,
       2007., 2010., 2013., 2004., 1990., 1988., 2002., 2001., 2011.,
       1998., 2015., 2012., 2014., 1992., 1997., 1993., 1994., 1982.,
       2016., 2003., 1986., 2000.,   nan, 1995., 1991., 1981., 1987.,
       1980., 1983.])

In [18]:
games['critic_score'].unique()

array([76., nan, 82., 80., 89., 58., 87., 91., 61., 97., 95., 77., 88.,
       83., 94., 93., 85., 86., 98., 96., 90., 84., 73., 74., 78., 92.,
       71., 72., 68., 62., 49., 67., 81., 66., 56., 79., 70., 59., 64.,
       75., 60., 63., 69., 50., 25., 42., 44., 55., 48., 57., 29., 47.,
       65., 54., 20., 53., 37., 38., 33., 52., 30., 32., 43., 45., 51.,
       40., 46., 39., 34., 35., 41., 36., 28., 31., 27., 26., 19., 23.,
       24., 21., 17., 22., 13.])

In [19]:
games['user_score'].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

Видим, что первые две колонки фактически содержат целые числа, а третья - дробные числа. Приведём данные к подходящим типам.



In [33]:
columns = ['year_of_release', 'critic_score', 'user_score']

for name in columns:
    games[name] = pd.to_numeric(games[name], errors='coerce', downcast='unsigned')
    print(headline(name))
    display(games[name].unique())


=================YEAR_OF_RELEASE==================


array([2006., 1985., 2008., 2009., 1996., 1989., 1984., 2005., 1999.,
       2007., 2010., 2013., 2004., 1990., 1988., 2002., 2001., 2011.,
       1998., 2015., 2012., 2014., 1992., 1997., 1993., 1994., 1982.,
       2016., 2003., 1986., 2000.,   nan, 1995., 1991., 1981., 1987.,
       1980., 1983.])


===================CRITIC_SCORE===================


array([76., nan, 82., 80., 89., 58., 87., 91., 61., 97., 95., 77., 88.,
       83., 94., 93., 85., 86., 98., 96., 90., 84., 73., 74., 78., 92.,
       71., 72., 68., 62., 49., 67., 81., 66., 56., 79., 70., 59., 64.,
       75., 60., 63., 69., 50., 25., 42., 44., 55., 48., 57., 29., 47.,
       65., 54., 20., 53., 37., 38., 33., 52., 30., 32., 43., 45., 51.,
       40., 46., 39., 34., 35., 41., 36., 28., 31., 27., 26., 19., 23.,
       24., 21., 17., 22., 13.])


====================USER_SCORE====================


array([8. , nan, 8.3, 8.5, 6.6, 8.4, 8.6, 7.7, 6.3, 7.4, 8.2, 9. , 7.9,
       8.1, 8.7, 7.1, 3.4, 5.3, 4.8, 3.2, 8.9, 6.4, 7.8, 7.5, 2.6, 7.2,
       9.2, 7. , 7.3, 4.3, 7.6, 5.7, 5. , 9.1, 6.5, 8.8, 6.9, 9.4, 6.8,
       6.1, 6.7, 5.4, 4. , 4.9, 4.5, 9.3, 6.2, 4.2, 6. , 3.7, 4.1, 5.8,
       5.6, 5.5, 4.4, 4.6, 5.9, 3.9, 3.1, 2.9, 5.2, 3.3, 4.7, 5.1, 3.5,
       2.5, 1.9, 3. , 2.7, 2.2, 2. , 9.5, 2.1, 3.6, 2.8, 1.8, 3.8, 0. ,
       1.6, 9.6, 2.4, 1.7, 1.1, 0.3, 1.5, 0.7, 1.2, 2.3, 0.5, 1.3, 0.2,
       0.6, 1.4, 0.9, 1. , 9.7])

pandas.to_numeric

In [ ]:
В третьей также присутствует значение `'tbd'`, которое означает отстутсвие рейтин 

### Итоги обзора и исправления данных

- Первый;
- Второй;
- Третий.

## Расходы каждого пользователя в месяц

### Итоги подготовки информации о расходах:

- Первый;
- Второй;
- Третий.

## Анализ расходов

### Итоги анализа расходов

- Первый;
- Второй;
- Третий.

## Гипотезы и проверки

### Итоги проверки гипотез

- Первый;
- Второй;
- Третий.

## Общие итоги

- Исследовано:
  - выборка пяти сотен пользователей из разных регионов, подключившихся в 2018 году;
  - полная статистика их потребления звонков, коротких сообщений и интернет-трафика.
- **Подтверждённые выводы:**
  - Пользователь тарифа **"Ultra"** приносит **в среднем на 33% больше выручки**, чем пользователь тарифа "Smart".
  - **Средняя выручка** с пользователя **в Москве не отличается** от средней выручки с пользователя в других регионах.
- Также обнаружено, что:
  - Пользователей тарифа "Smart" в 4 раза больше;
  - Среди тех, кто потребляет много, популярней тариф "Ultra";
  - В среднем пользователи "Smart" потребляют столько, сколько включено в базовый пакет;
  - При относительно небольшом увеличении потребления их чек заметно вырастает;
  - Если потребление превышает 100 звонков (или 750 минут), или 110 сообщений, или 25 ГБ в месяц - пользователи предпочитают тариф "Ultra";
  - Пользователи "Ultra" потребляют только треть от базового пакета;
  - И 85% из них платят только базовый платёж.
- Предположения:
  - Пользователи "Smart":
    - сознательно ограничивают своё потребление;
    - при повышении потребления переходят на гораздо более содержательный "Ultra".
  - Пользователи "Ultra":
    - при повышении потребления вдвое не станут платить вдвое больше.

### Рекомендации

**Описание**

- пишем от лица "мы", употребляем подходящие времена
- Пиши краткие промежуточные выводы.
- Сделай общий вывод по итогам проекта в отдельном разделе.
- Сохрани идеи и код, который помогает принимать промежуточные решения.
- удали неработающий код
- комментарии ставь перед кодом
- вынеси комментарии в маркдаун
- подробно описывай процесс, что и почему ты делаешь
- за каждым пояснением - ячейка с кодом. Одна ячейка - один шаг
- перед отправкой - перезапусти вессь проект, убедись, что все ячейки сработали верно

**Вывод данных**
- Каждому объекту данных (фрейм, график) - дай название, подпись, подпиши столбцы и оси.
- Таблицы давай выборками, .head() и .tail()
- таблицы выводи по display()  # from IPython.display import display

**Итоги**
- краткий обзор проведённой работы: что выполнили, рассчитали:
- главные выводы: ответы на поставленные вопросы
- рекомендации: как избежать неправильных данных, как отобрать признаки для прогнозирования в будущем

На что обращают внимание при проверке проекта:
- [ ] Как вы описываете выявленные в данных проблемы?
- [ ] Как готовите датасет к анализу?
- [ ] Какие строите графики для распределений и как их объясняете?
- [ ] Как рассчитываете стандартное отклонение и дисперсию?
- [ ] Формулируете ли альтернативную и нулевую гипотезы?
- [ ] Какие методы применяете, чтобы их проверить?
- [ ] Объясняете результат проверки гипотезы или нет?
- [ ] Соблюдаете ли структуру проекта и поддерживаете аккуратность кода?
- [ ] Какие выводы делаете?
- [ ] Оставляете ли комментарии к шагам?